test/test.js

```diff
@@ -23,7 +23,7 @@ request.Test.prototype.expectJSON = function(json, done) {
 request.Test.prototype.expectNoHeader = function(header, done) {
   this.expect(function(res) {
     if (header.toLowerCase() in res.headers) {
-      return 'Unexpected header in response: ' + header;
+      return Error('Unexpected header in response: ' + header);
```

Comment:
Could you prepend the `new` keyword before the `Error` constructor?

Reply:
revised single commit pushed